plugins/floatpanel/plugin.js

```diff
@@ -143,7 +143,20 @@ CKEDITOR.plugins.add( 'floatpanel', {
 			 */
 			showBlock: function( name, offsetParent, corner, offsetX, offsetY, callback ) {
 				var panel = this._.panel,
-					block = panel.showBlock( name );
+					block = panel.showBlock( name ),
+					rectList,
+					rect,
+					rtl = this._.dir == 'rtl';
+				if ( offsetX === undefined || offsetY === undefined ) {
+					rectList = this._.editor.getSelection().getRanges()[ 0 ].getClientRects();
```

Comment:
It would be better to have also fallback for case when `getClientRects` return empty array (e.g. due to w3c/csswg-drafts#2514). Because now this code fails for programatically opened context menus:

```javascript
var editor = CKEDITOR.instances.editor,
  range = editor.createRange(),
  element = editor.editable().getFirst();

range.setStart( element, CKEDITOR.POSITION_AFTER_START );
range.select();
editor.execCommand( 'contextMenu' );
```

When getting position of the selection is not possible, falling back to opening menu in position (0,0) seems reasonable.

Reply:
After recent changes to getClientRects it's highly unlikely to get empty array as rect list, however I will add this fallback in case of future changes to our or native getClientRects.